tensorflow/java/src/main/java/org/tensorflow/types/Types.java

```diff
@@ -0,0 +1,37 @@
+package org.tensorflow.types;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.tensorflow.DataType;
+
+/**
+ * Utility class for representing TF types as Java types. For each TF type (e.g., int32),
+ * there is a corresponding Java type (e.g., TFInt32) that represents it at compile time
+ * and a corresponding immutable object (e.g., TFInt32.T) that represents it at run time.
+ */
+public class Types {
+
+  private Types() {} // not instantiable
+
+  static final Map<Class<?>, Integer> typeCodes = new HashMap<>();
+  /** Convert to the equivalent DataType. */
+
+  public static DataType dataType(Class<?> c) {
```

Comment:
Should this be `Class<? extends TFType>`?
(In which case it won't be possible to have a runtime exception).

Or is this because we were (are?) thinking of `Tensor<Integer>` and thus the needed to use `Types.dataType(Integer.class)` as well?